include/envoy/filesystem/watcher.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
```

Comment:
I'm not sure what's possible, but is there any chance to split some of these changes out into a separate PR? Meaning, can we do all the interface work in one PR, and then come back and add Windows? I think that would be easier to review.

Reply:
I think it's worth it to do the interface changes separate from the windows stuff, if you don't mind.

Reply:
sure, we can back the windows changes out of this PR